core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -162,6 +161,25 @@ public ClusterState execute(ClusterState currentState) throws Exception {
         });
     }
 
+    void validatePipeline(Map<DiscoveryNode, IngestInfo> ingestInfos, PutPipelineRequest request) throws Exception {
```

Comment:
and then we end up in state where the put pipeline API acknowledge the request, but one or more nodes aren't able to use the ingest pipeline and this should be prevented.